### ISSUE TITLE
String Data Size Limit

### ISSUE THREAD

**Issue by mahasanti** [jeog/TOSDataBridge#27]
Is there a way to adjust the maximum string data size, so that I can access a larger string custom quotes?

I have been working on accessing data through the custom quotes. I have been successful, however I am limited by the 40 character limit of the STR_DATA_SZ constant. My custom quote data is therefore prematurely truncated. TOS is able to export much larger strings, which I have verified with excel RTD functions. 
 Would I have to change TOSDB_STR_DATA_SZ in tos_databridge.h and recompile?

Thank you.

**Comment by jeog** [comment]
Yes, (re)defining TOSDB_STR_DATA_SZ *should* allow that. But, I noticed a while back that when building on new versions of VS a few things now break (I assume there was some non standard compliant stuff they fixed.) So, you'll either need to use an earlier version of VS/toolset or make a few changes in the code to get it to compile.

**Comment by jeog** [comment]
I pushed branch [fixes-v142](https://github.com/jeog/TOSDataBridge/tree/fixes-v142) with some fixes that get this to build on vs2017(v141) and vs2019(v142) although i didn't have time to test the changes.

Also, if you do change the size of str data keep in mind the streaming data is stored in circular buffers of size TOSDB_SHEM_BUF_SZ (4096) and gets overwritten once full.  So, if you up the str data size to 1024 you will only get 4 entries in there before it gets over-written. You *should* be able to increase this size as well, if needed.

**Comment by mahasanti** [comment]
Thank you, this looks great. I am excited to try this out. I will test it out and let you know if I see any issues. 
Cheers.

**Comment by mahasanti** [comment]
I have been testing this out and was able to increased the string size to 255. Increasing TOSDB_STR_DATA_SZ > 255 didn't increase the limit. I also changed TOSDB_STR_DATA_SZ, TOSDB_MAX_STR_SZ, to no avail. I am guessing there is someplace that limits string length to 1 byte, or something. Or maybe a buffer limit somewhere.  I poked around for a while, but couldn't see where the limit was.

Honestly, a limit of 255 is much better, and I think I can work with this. Let me know if anything comes to mind about what could be limiting this. 

Thanks again for an awesome project!
